SonarTfsAnnotate/HistoryProvider.cs

```diff
@@ -44,25 +45,34 @@ public HistoryProvider(IEnumerable<Changeset> changesets)
 
         public bool Next()
         {
-            if (current - 1 >= 0)
+            while (true)
             {
-                Dispose(current - 1);
-            }
+                if (current - 1 >= 0)
+                {
+                    Dispose(current - 1);
+                }
 
-            current++;
-            if (current >= changesets.Count)
-            {
-                return false;
-            }
+                current++;
+                if (current >= changesets.Count)
+                {
+                    return false;
+                }
 
-            if (current + PREFETCH_SIZE < changesets.Count)
-            {
-                Prefetch(current + PREFETCH_SIZE);
-            }
+                if (current + PREFETCH_SIZE < changesets.Count)
+                {
+                    Prefetch(current + PREFETCH_SIZE);
+                }
+
+                manualResetEvents[current].WaitOne();
 
-            manualResetEvents[current].WaitOne();
+                if (!File.Exists(filenames[current]))
+                {
+                    // The download was not successful. Move on to the next file.
+                    continue;
```

Comment:
ChangeType: add, source rename can be obtained in the following series of steps
1. Checkin a File - File1
2. Rename it to File2 and add a new file named File1. 
3. Checkin the pending changes

Now, File1 History will have the aforementioned ChangeType
In the case of this example, both Visual Studio and TFS Annotate are able to annotate the file, and so, the problem may not necessarily lie with the specific ChangeType.